mantidproject/mantid
# Updating sans GUI table

**Description of work.**

This is the initial work to replace the DataProcessorWidget with the JobTreeView widget in the new Sans GUI. This has involved doing the following:
* Updating all the methods on the view to interact with the JobTreeView rather than the DataProcessorWidget
* Adding a removeAllRows method to JobTreeView
* Updated the table model to keep an up to date version of the table in the view. 
* In main_presenter.py removed the inheritance from DataProcessorMainPresenter as well as all the now unneeded code. This class is now basically uneeded and will probably be removed in a future branch.
* Updating run_tab_presenter to interact with the new table_view and table_model as well as run the reduction process itself.





**To test:**
* Attempt to run a SANS reduction in the new GUI
* Play around and check usability is good. 


Fixes #23022 . 

Does this update require release notes?


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
